src/v/kafka/server/handlers/describe_configs.cc

```diff
@@ -77,12 +78,42 @@ static ss::sstring describe_as_string(const T& t) {
     return ssx::sformat("{}", t);
 }
 
+
+// property_config_type maps the datatype for a configuration to describe_configs_type.
+// Currently class_type and password are not used in Redpanda so we do not include checks for those types.
+// You may find a similar mapping in Apache Kafka at https://github.com/apache/kafka/blob/be032735b39360df1a6de1a7feea8b4336e5bcc0/core/src/main/scala/kafka/server/ConfigHelper.scala
+// Type sizes are based on Java and taken from https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html
+template<typename T>
+describe_configs_type property_config_type() {
+  // Default type is unknown
+  describe_configs_type type = describe_configs_type::unknown;
+
+  if constexpr (std::is_same_v<T, bool>) {
+    type = describe_configs_type::boolean;
+  } else if constexpr (std::is_same_v<T, ss::sstring>) {
+    type = describe_configs_type::string;
+  }else if constexpr (std::is_same_v<T, int32_t>) {
```

Comment:
what about unsigned types? Check out `std::is_integral`

Reply:
I agree that I should account for unsigned types but I question the need for `is_integral`.

`is_integral` returns true for **any** integral type. I think that is an issues because an unsigned 16-bit integer and an unsigned 32-bit integer could all map to the same config_type.

Reply:
yeah i was thinking something like:
```
if (is integral<T>){
    if( is_long<T>) {
        ....
    } if(is_short<T>) {
       ....
       etc
```

Reply:
> yeah i was thinking something like:
> 
> ```
> if (is integral<T>){
>     if( is_long<T>) {
>         ....
>     } if(is_short<T>) {
>        ....
>        etc
> ```

I missed this comment. I'll update in the next push

Reply:
> > yeah i was thinking something like:
> > ```
> > if (is integral<T>){
> >     if( is_long<T>) {
> >         ....
> >     } if(is_short<T>) {
> >        ....
> >        etc
> > ```
> 
> I missed this comment. I'll give it a try after the next round of reviews.

After getting some rest, I don't see a difference between the suggestion and something like:

```cpp
if (is_same_v<T, int32_t> || is_same_v<T, uint32_t>) {
...
} ...
```

IMO the above is more clear since it explicitly accounts for the bit size of a type.

Reply:
Theres other situations like `int` that we should probably account for though.